lib/internal/test_runner/test.js

```diff
@@ -183,6 +185,11 @@ class Test extends AsyncResource {
         return;
       }
 
+      if (this.shouldReportSubtest) {
+        this.reporter.subtest(this.indent, this.name);
```

Comment:
It still might be possible to move this to `finalize()`. I think the logic would look something like:

```js
if (this.parent.waitingOn === 0) {
  this.reporter.subtest(this.parent.indent, this.parent.name);
}
```

Reply:
well, as you mentioned, this comment https://github.com/nodejs/node/blob/72f9597d3e8f0360075d45a2114b7c3670668fad/lib/internal/test_runner/test.js#L408

the finalize of subtests has already ran meaning this is basically dfs - you must print this *before* calling `finalize`
I tried running in the `run` method but that created this issue https://github.com/nodejs/node/pull/43417#discussion_r897021968